src/extension_commands.ml

```diff
@@ -534,6 +534,116 @@ module Copy_type_under_cursor = struct
+        `Warn
+        "The installed version of `ocamllsp` does not support type search. %s"
+        msg
+
+  let _get_search_results ~query ~limit ~with_doc text_editor client =
+    let doc = TextEditor.document text_editor in
+    let uri = TextDocument.uri doc in
+    let selection = TextEditor.selection text_editor in
+    Custom_requests.(
+      send_request
+        client
+        Type_search.request
+        (Type_search.make
+           ~uri
+           ~position:(Range.start (Selection.to_range selection))
+           ~limit
+           ~query
+           ~with_doc
+           ()))
+
+  let _search_by_type =
+    let handler (instance : Extension_instance.t) ~args:_ =
+      let search_by_type () =
+        match Window.activeTextEditor () with
+        | None ->
+          Extension_consts.Command_errors.text_editor_must_be_active
+            extension_name
+            ~expl:"The command to search for a value by it's type"
+          |> show_message `Error "%s" |> Promise.return
+        | Some _text_editor -> (
+          match Extension_instance.lsp_client instance with
+          | None ->
+            show_message `Warn "ocamllsp is not running" |> Promise.return
+          | Some (_client, ocaml_lsp)
+            when not (Ocaml_lsp.can_handle_search_by_type ocaml_lsp) ->
+            ocaml_lsp_doesnt_support_search_by_type instance ocaml_lsp
+            |> Promise.return
+          | Some (_client, _) ->
+            let _get_query_input =
+              Vscode.(
+                Window.showInputBox
+                  ~options:
+                    (InputBoxOptions.create
+                       ~title:"Search By Type"
+                       ~prompt:
+                         "Perform a search by type request by providing a type \
+                          signature to look for"
+                       ~placeHolder:"int -> string"
+                       ~value:"int -> string"
+                       ~password:false
+                       ~ignoreFocusOut:true
+                       ())
+                  ())
+            in
+            let _get_doc_input =
+              Vscode.(
+                Window.showQuickPickItems
+                  ~choices:
+                    [ ( QuickPickItem.create ~label:"Yes" ~alwaysShow:false ()
+                      , "yes" )
+                    ; ( QuickPickItem.create ~label:"No" ~alwaysShow:false ()
+                      , "no" )
+                    ]
+                  ~options:
+                    (QuickPickOptions.create
+                       ~title:"Search By Type"
+                       ~matchOnDescription:false
+                       ~canPickMany:false
+                       ~placeHolder:
+                         "Documentation information will be attached for each \
+                          result."
+                       ~ignoreFocusOut:true
+                       ())
+                  ())
+            in
+            let _get_limit_input =
```

Comment:
At the moment, if you disable the boolean check above in the [comment](https://github.com/ocamllabs/vscode-ocaml-platform/pull/1626/files#r1808497434) and run, only this input displays. 

To have a series of inputs we need to implement a way to run each one successively after the other while retaining the results of each step.

An example of this in typescript can be found at: https://github.com/microsoft/vscode-extension-samples/blob/main/quickinput-sample/src/multiStepInput.ts#L169

Reply:
Yes, but do we really need all these ?

I expect only one input to shows: the actual query that the user needs to enter.

If we feel like we want to make other parameter configurable, we can have real options for that.

Anyway, there is an issue right now since the request is performed without waiting fro the user input.

Reply:
This works great.